File: jar-common/src/main/java/com/beancounter/common/contracts/AssetSearchResult.java
package com.beancounter.common.contracts;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class AssetSearchResult {
  private String symbol;
  private String name;
  private String type;
  private String region;
}

File: jar-contracts/src/main/kotlin/com/beancounter/common/contracts/AssetSearchResult.kt
package com.beancounter.common.contracts

/**
 * The found asset.
 */
data class AssetSearchResult(
    var symbol: String,
    var name: String,
    var type: String,
    var region: String?,
    var currency: String?,
    var market: String? = null,
    var assetId: String? = null
)